cms/grading/ScoreType.py

```diff
@@ -655,3 +651,88 @@ class IOIScoreTypeGroup(AIOCScoreTypeGroup):
     </div>
 </div>
 {% end %}"""
+
+    def compute_score(self, submission_result):
+        """See AIOCScoreTypeGroupScoreType.compute_score."""
+        # Actually, this means it didn't even compile!
+        if not submission_result.evaluated():
+            return 0.0, "[]", 0.0, "[]", \
+                json.dumps(["%lg" % 0.0 for _ in self.parameters])
+
+        # Lexicographical order by codename
+        evaluations = dict((ev.codename, ev)
+                           for ev in submission_result.evaluations)
+        subtasks = []
+        public_subtasks = []
+        ranking_details = []
+
+        for st_idx, parameter in enumerate(self.parameters):
+            st_max_score = parameter[1]
+            st_score = self.reduce([float(evaluations[idx].outcome)
+                                    for idx in parameter[2]],
+                                   parameter) * parameter[1]
+            st_public = all(self.public_testcases[idx]
+                            for idx in parameter[2])
+            st_sample = (st_max_score <= 0)
+
+            tc_outcomes = dict((
+                idx,
+                self.get_public_outcome(
+                    float(evaluations[idx].outcome), parameter)
+                ) for idx in parameter[2])
+
+            testcases = []
+            public_testcases = []
+
+            for idx, testcase_id in enumerate(parameter[2]):
+                if (st_sample or idx == len(parameter[2]) - 1 or
+                    tc_outcomes[testcase_id] != "Correct"):
+                    testcase_details = {
```

Comment:
Does this successfully run? I thought python indent rules meant you had to double-indent the continuation clause (or double-indent the block)